### PR TITLE
feat(19715): integrate breadcrumbs component into drug flow

### PR DESCRIPTION
**Pull request by Natallia-Harshunova**
https://kontur.fibery.io/Tasks/Task/Integrate-map-breadcrumbs-component-into-drag-flow-19715

## Release Notes

- **New Features**
  - Introduced a "Zoom to the Whole World" functionality in the map interface.
  - Added a new `BreadcrumbsPanel` component for enhanced navigation, conditionally rendered based on feature flags.

- **Improvements**
  - Enhanced error handling and logic flow for breadcrumb item management.
  - Updated dropdown selection logic for better user experience.

- **Bug Fixes**
  - Adjusted interaction logic for the crosshair element to improve usability.

- **Styles**
  - Updated CSS for the crosshair to disable pointer events, allowing for better interaction with underlying elements.

- **Documentation**
  - Improved localization strings for better user experience.

### CODE REVIEW

File: src/features/breadcrumbs/BreadcrumbsPanel.tsx [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
+import {
+  breadcrumbsItemsAtom,
+  onBreadcrumbClick,
+  onZoomToWholeWorld,
+} from './atoms/breadcrumbsItemsAtom';
+
+const noBreadcrumbsOption = {
+  label: i18n.t('no_breadcrumbs_label'),
+  value: 'no breadcrumb items',
+};
```
[the remark this answers is not in the document]
Comment (Natallia-Harshunova) [reply src/features/breadcrumbs/BreadcrumbsPanel.tsx]: what do you mean?

File: src/core/api/boundaries.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { apiClient } from '~core/apiClientInstance';
+import { GeoJSONPoint } from '~utils/geoJSON/helpers';
+
+export function getBoundaries(
+  coords: [number, number],
+  abortController?: AbortController,
+) {
+  return apiClient.post<GeoJSON.FeatureCollection>(
+    '/boundaries',
```
Comment (albaranau) [inline src/core/api/boundaries.ts:9]: Maybe we want an alternative endpoint here? `/boundaries` returns GeoJSONs, and they can be pretty heavy
image

upd: it's probably okay for now, if it's an issue, we should create a separate ticket

File: src/core/localization/translations/en/common.json [hunk 3]
```diff
@@ -435,6 +435,7 @@
   },
   "episode": "Timeline",
   "loading_episodes": "Loading Episodes",
+  "no_breadcrumbs_label": "Zoom to the whole world",
```
Comment (albaranau) [inline src/core/localization/translations/en/common.json]: Maybe "zoom_to_world" (or similar) would be a better key? It's a string for "zoom to the whole world" button, not for info "no breadcrumbs" label.

**Comment by Komzpa** [comment]
> drug flow
image

File: src/features/breadcrumbs/BreadcrumbsPanel.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Komzpa) [reply src/features/breadcrumbs/BreadcrumbsPanel.tsx]: @coderabbitai remember that we are supporting users dragging map over an ocean to view the cyclone details and trying to zoom out of the blue map. The experience inside "no country" still should allow navigation.

File: src/features/breadcrumbs/atoms/breadcrumbsItemsAtom.ts [hunk 4]
```diff
@@ -1,12 +1,19 @@
...
-    breadcrumbsItemsAtom(ctx, items.slice(0, index + 1));
-  }
-}, 'onItemClick');
+  if (!items) return;
+  const index = items.findIndex((item) => item.id === value);
+}, 'onBreadcrumbClick');
+
+export const onZoomToWholeWorld = action(() => {
+  const bbox: Bbox = [-180, -80, 180, 80];
+  currentMapPositionAtom.setCurrentMapBbox.dispatch(bbox);
```
[the remark this answers is not in the document]
Comment (Komzpa) [reply src/features/breadcrumbs/atoms/breadcrumbsItemsAtom.ts:28]: @coderabbitai remember that we use the map in spherical mercator (EPSG:3857) where the poles are not representable

File: src/features/breadcrumbs/BreadcrumbsPanel.tsx [hunk 5]
```diff
@@ -1,17 +1,36 @@
...
 import s from './BreadcrumbsPanel.module.css';
-import { breadcrumbsItemsAtom, onBreadcrumbClick } from './atoms/breadcrumbsItemsAtom';
+import {
+  breadcrumbsItemsAtom,
+  onBreadcrumbClick,
+  onZoomToWholeWorld,
+} from './atoms/breadcrumbsItemsAtom';
+
+const noBreadcrumbsOption = {
+  label: i18n.t('no_breadcrumbs_label'),
```
Comment (vkozio) [inline src/features/breadcrumbs/BreadcrumbsPanel.tsx]: i18n.t('zoom_to_the_whole_world') please.
Translation keys are not perfect, but let's not make them worse.

File: src/features/breadcrumbs/BreadcrumbsPanel.tsx [hunk 6]
```diff
@@ -1,17 +1,36 @@
...
+    ? constructOptionsFromBoundaries(items || new FeatureCollection([]))
+    : [noBreadcrumbsOption];
+  const clickHandler = items?.length ? breadcrumbItemClick : zoomToTheWorld;
 
-  return items.length > 0 ? (
+  /** Don't render the Breadcrumbs panel or empty label until items have been fetched from the server. */
+  return items ? (
     <Panel resize="none" className={s.breadcrumbsPanel}>
-      <Breadcrumbs items={items} onClick={handleClick} />
+      <Breadcrumbs items={options.reverse()} onClick={clickHandler} />
```
[the remark this answers is not in the document]
Comment (vkozio) [reply src/features/breadcrumbs/BreadcrumbsPanel.tsx]: we have .toReversed() but I'd suggest to create options array in proper order initially

**Review by vkozio — CHANGES_REQUESTED** [review]
marked relevant rabbit comments, please pay attention

File: src/core/api/boundaries.ts (hunk 2 above)
Comment (Natallia-Harshunova) [reply src/core/api/boundaries.ts:9]: Yes, let's bring this discussion to the next meeting, but for this pr leave it as it is

**Review by amurKontur — APPROVED** [review]

**Review by vkozio — APPROVED** [review]

**Pull request merged**